test/hack/resource/pkg/resourcetypes/resourcetypes.go

```diff
@@ -34,6 +34,8 @@ const (
 type Type interface {
 	// String is the string representation of the type
 	String() string
+	// Region returns the region the resource is located
+	Region() string
```

Comment:
```suggestion
	Global() bool
```

Just a thought. I'm fine either way. But this would mean that we wouldn't have to pass the region down into every resourcetype client and can just check if something is a global resource or not